packages/core/contracts/chainbridge/BeaconOracle.sol

```diff
@@ -0,0 +1,150 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+pragma solidity ^0.6.0;
+pragma experimental ABIEncoderV2;
+
+import "../oracle/interfaces/FinderInterface.sol";
+import "./IBridge.sol";
+import "../oracle/implementation/Constants.sol";
+import "../oracle/interfaces/OracleAncillaryInterface.sol";
+
+/**
+ * @title Simple implementation of the OracleInterface used to communicate price request data cross-chain between
+ * EVM networks. Can be extended either into a "Source" or "Sink" oracle that specializes in making and resolving
+ * cross-chain price requests, respectivly. The "Source" Oracle is the originator or source of price resolution data
+ * and can only resolve prices already published by the DVM. The "Sink" Oracle receives the price resolution data
+ * from the Source Oracle and makes it available on non-Mainnet chains. The "Sink" Oracle can also be used to trigger
+ * price requests from the DVM on Mainnet.
+ */
+abstract contract BeaconOracle is OracleAncillaryInterface {
```

Comment:
This is way simpler! +1

Reply:
Than what?

Reply:
Sorry, I just meant the entire contract is way simpler without the need to be able to list all the requests (the refactor you did!).